src/cart/tests_common.h

```diff
@@ -104,6 +108,35 @@ tc_progress_stop(void)
 	opts.shutdown = 1;
 }
 
+/* Write a completion file to signal graceful server shutdown */
+void
+write_completion_file(void)
+{
+	FILE	*fptr;
+	char	*dir;
+	char	*completion_file;
+	char	*tmp_str;
+	char	pid[6];
+	pid_t	_pid;
+
+	_pid = getpid();
+	sprintf(pid, "%d", _pid);
+
+	dir = getenv("DAOS_TEST_SHARED_DIR");
+	tmp_str = strcat(dir, "/test-servers-completed.txt.");
+	completion_file = strcat(tmp_str, pid);
+
+	fptr = fopen(completion_file, "w");
```

Comment:
what if completion file already exists? should we delete it first?

Reply:
If the file exists with matching PID, it means either the server called `write_completion_file()` twice or avocado left a very old completion file around.  I say we unlink it beforehand, so that the timestamp is set to the last call to `write_completion_file()`.

Reply:
sounds good

Reply:
984ab8326148b1094ecf16bd2e180f90ebf80c23